src/main/java/ru/herobrine1st/fusion/module/vk/util/Extensions.kt

```diff
@@ -53,6 +54,21 @@ fun Post.toEmbeds(wallName: String, wallAvatarUrl: String?, repost: Boolean = fa
             }
             if (size > 4) footerBuilder.append("Post contains more than 4 images\n")
         }
+        with(attachments.filterIsInstance<Document>()) {
```

Comment:
It should not add another 4 images, better to map both lists to URLs and merge them while keeping original order, and only then `subList`.

Something like

```kotlin
val urls = emptyList<String>()
attachments.forEach {
  if(it is Photo) { urls += it.getLargestSize().url }
  else if(it is Document && it.type == GIF) { urls += it.url }
}
with(urls) {
  if(isEmpty) return@with
  embedBuilder.setImage(this.first())
  this.subList(1, min(4, size)).forEach {
    embeds.add(
      ModifiedEmbedBuilder()
        .setTitle(null, url)
        .setImage(it)
        .build()
      )
  }
}
```